karton/core/karton.py

```diff
@@ -6,16 +6,19 @@
 import sys
 import time
 import traceback
-from typing import Any, Callable, Dict, List, Optional, Tuple, cast
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union, cast
 
 from .__version__ import __version__
 from .backend import KartonBackend, KartonBind, KartonMetrics
 from .base import KartonBase, KartonServiceBase
 from .config import Config
+from .exceptions import TaskTimeoutError
 from .resource import LocalResource
 from .task import Task, TaskState
 from .utils import timeout
 
+KartonTaskException = Union[Exception, BaseException]
```

Comment:
:thinking: BaseException is base class of Exception. Isn't just BaseException good enough as a type?